pep517/wrappers.py

```diff
@@ -210,11 +218,19 @@ def _call_hook(self, hook_name, kwargs):
                               indent=2)
 
             # Run the hook in a subprocess
-            self._subprocess_runner(
-                [sys.executable, _in_proc_script, hook_name, td],
-                cwd=self.source_dir,
-                extra_environ=extra_environ
-            )
+            if _HAVE_IMPORTLIB_RESOURCES:
+                with resources.path(__package__, '_in_process.py') as script:
+                    self._subprocess_runner(
```

Comment:
To avoid duplicating the calls to `_subprocess_runner`, I would define a context manager in each branch of the `try: ... except ImportError: ...` itself that yields the expected path, like

```python
try:
    import importlib.resources as resources

    _in_proc_script_path = lambda: resources.path(__package__, '_in_process.py')
except ImportError:
    from contextlib import contextmanager

    _in_proc_script_path = contextmanager(
        lambda: pjoin(dirname(abspath(__file__)), '_in_process.py')
    )
```

then use `with _in_proc_script_path() as script` unconditionally.